src/js/components/FormField/FormField.js

```diff
@@ -526,6 +549,19 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
+          const targetType = event.target.type || event.target.tagName;
+          // Check if the target is one of the input
+          // components that have their own focus indicator
+          const shouldSkipFocus = [
+            'checkbox', // For CheckBox
+            'radio', // For RadioButton
+            'range', // For RangeInput
+          ].includes(targetType.toLowerCase());
+
+          // skip focus
+          if (shouldSkipFocus && theme.formField.focusIndicator === false) {
+            return;
```

Comment:
It seems like the onFocus event should still trigger we just shouldn't get the styling